source/code/plugins/npmd_config_lib.rb

```diff
@@ -244,41 +250,175 @@ def self.createActOnElements(actOnArray, subnetIdHash, xmlElemName)
+                
+                if _rule["NetworkTestMatrix"].empty?
                     Logger::logWarn "Skipping rule #{x["Name"]} as network test matrix is empty", Logger::loop
                     @@rule_drops += 1
                 else
-                    _alertConfig = REXML::Element.new("AlertConfiguration")
-                    _alertConfig.add_element("Loss", {"Threshold" => x["LossThreshold"] })
-                    _alertConfig.add_element("Latency", {"Threshold" => x["LatencyThreshold"]})
-                    _exceptions = createActOnElements(x["Exceptions"], subnetIdHash, "Exceptions")
-                    _rule.elements << _alertConfig
-                    _rule.elements << _netTestMtx
-                    _rule.elements << _exceptions
-                    _rules.elements << _rule
+                    # Alert Configuration
+                    _rule["AlertConfiguration"]["ChecksFailedPercent"]  = x["LossThreshold"]
+                    _rule["AlertConfiguration"]["RoundTripTimeMs"]  = x["LatencyThreshold"]
+                end
+                if !_rule.empty?
+                    _rules.push(_rule)
                 end
             end
             _rules
         end
+
+        def self.createEpmElements(epmHash)
+            _epmRules = Hash.new
+            _rule = Array.new
+            epmHash.each do |key, rules|
+                for i in 0..rules.length-1
+                    _ruleHash = Hash.new
+                    _iRule = rules[i] # get individual rule
+                    _ruleHash["ID"] = _iRule["ID"]
+                    _ruleHash["Name"] = _iRule["Name"]
+                    _ruleHash["CMResourceId"] = _iRule["CMResourceId"]
+                    _ruleHash["IngestionWorkspaceId"] = _iRule["IngestionWorkspaceId"]
+                    _ruleHash["WorkspaceAlias"] = _iRule["WorkspaceAlias"]
+                    _ruleHash["Redirect"] = "false"
+                    _ruleHash["NetTests"] = (_iRule.has_key?("NetworkThresholdLoss") and _iRule.has_key?("NetworkThresholdLatency")) ? "true" : "false"
+                    _ruleHash["AppTests"] = (_iRule.has_key?("AppThresholdLatency")) ? "true" : "false"
+                    if (_ruleHash["NetTests"] == "true")
+                        _ruleHash["NetworkThreshold"] = {"ChecksFailedPercent" => _iRule["NetworkThresholdLoss"], "RoundTripTimeMs" => _iRule["NetworkThresholdLatency"]}
+                    end
+
+                    if (_ruleHash["AppTests"] == "true")
+                        _ruleHash["AppThreshold"] = {"ChecksFailedPercent" => _iRule["AppThresholdLoss"], "RoundTripTimeMs" => _iRule["AppThresholdLatency"]}
+                    end
+
+                    # Fill endpoints
+                    _epList = _iRule["Endpoints"]
+                    _endpointList = Array.new
+                    for j in 0.._epList.length-1
+                        _epHash = Hash.new
+                        _epHash["Name"] = _epList[j]["Name"]
```

Comment:
In general, what happens when you access an element when its not present like: 
_epList[j]["Name"], as SCM config doesnt has Name field. You need to check this for all the new fields added : Name, AppLoss, CMResourceId, etc.

Reply:
Validation added for new fields. In case the element is not present, it's value in config will be marked as empty string.